src/coreclr/vm/gcenv.ee.cpp

```diff
@@ -1090,16 +1105,32 @@ void GCToEEInterface::StompWriteBarrier(WriteBarrierParameters* args)
         g_region_to_generation_table = args->region_to_generation_table;
         g_region_shr = args->region_shr;
         g_region_use_bitwise_write_barrier = args->region_use_bitwise_write_barrier;
+        g_ephemeral_low = args->ephemeral_low;
+        g_ephemeral_high = args->ephemeral_high;
+#if defined(HOST_ARM64)
+        // Only allow bitwise write barriers if LSE atomics are present
+        if (!g_arm64_atomics_present)
+        {
+            g_region_use_bitwise_write_barrier = false;
+        }
+        // Skip ephemeral checks for regionless server GC
+        if (GCHeapUtilities::IsServerHeap() && g_region_to_generation_table == nullptr)
+        {
+            g_ephemeral_low = 0;
```

Comment:
Note that `g_ephemeral_low/high` are used outside of the asm helpers as well. I do not think that you can just set both to 0. I think it will break other uses.

Reply:
> Note that g_ephemeral_low/high are used outside of the asm helpers as well.

Fixed up. Note I've kept it so it doesn't do one call to `StompWriteBarrierEphemeral()` that isn't needed